sdks/java/core/src/main/java/org/apache/beam/sdk/io/fs/ResourceIdTester.java

```diff
@@ -104,6 +104,7 @@ private static void validateResolvingIds(
     //   May need options to be filesystem-independent, e.g., if filesystems ban certain chars.
   }
 
+  @SuppressWarnings("TryFailThrowable")
```

Comment:
This looks like the actual bug, rather than a spurious warning - the `AssertionError` thrown by fail can't ever be propagated out of the catch. You should instead fix the bug.

http://errorprone.info/bugpattern/TryFailThrowable

Reply:
makes sense, will do